docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,184 @@
+
+Glides MAY publish an event that assigns a vehicle to a trip even if the last update for that vehicle and trip assigned them to each other. Glides might do this to update the `carNumbers` associated with a `vehicleId`.
+
+Glides MUST NOT refer to the same vehicle or trip more than one time within all the assignments in one event.
+
+## Rules for Assignments
+
+This schema provides a method for saying which trains are assigned to which trips, but does not describe rules for when Glides will say a train is assigned to a trip. This section is a non-normative description of how Glides intends to decide which assigments to publish, but Glides may decide to change these rules and that would not be a breaking change to this schema, even if it has semantic consequences for consumers.
+
+This section may not be kept up to date as Glides' logic is updated.
+
+### Inputs
+
+Glides will consider as input to its trip matching process:
+
+- The schedule it gets from HASTUS
+- Data entered on trainsheets in Glides
+- Realtime train location data from RTR, including the direction and AVI code
+
+Getting direction from RTR may lead to a circular dependency if RTR gets its direction from Glides' trip assignments. RTR should therefore ignore this feed's trip assignments if they conflict with the direction RTR would otherwise assign to a train.
+
+Glides will do its best to make trip assignments even for trains that inspectors have not entered into the trainsheet, though the accuracy may decrease in that situation.
+
+### Layovers
+
+At all terminals, as soon as Glides learns that a train has reached the terminal, Glides will unassign the vehicle from the completed trip. If Glides knows which trip the vehicle will do next, it will immediately assign it to the next trip, otherwise, the vehicle will not be assigned to a trip. (This also applies at some non-terminals if a train starts its next trip immediately from the same platform, such as Government Center's Brattle Loop during disruptions, and at East Somerville when pullouts switch from a nonrevenue trip to a revenue trip for the rest of the pullout to Medford.)
+
+If Glides expects a train to turn around at a station in the middle of the line at a station where trains must loop to a different platform (Government Center Eastbound to Westbound, North Station, Kenmore, Park Street), when the train reaches the last stop, Glides will leave the train assigned to the previous trip. Glides will not unassign it from the previous trip and assign the train to the next trip until the train has actually turned around onto track that goes the other direction. This is so that if a train is unexpectedly extended (e.g. continuing from Government Center until North Station), the train won't be unassigned from a trip until it _actually_ finishes the trip.
```

Comment:
**nitpick:** I think it would actually be safe to do the unassign-upon-arrival approach for trains looping at Park Street because once a train is on the fence track there it physically has to turn around.

Reply:
(That's a weird edge case kind of nitpick, though, just something I wanted to point out without saying that means we should necessarily change the approach.)

Reply:
Good point. Except Glides currently ignores the child stop id and only looks at the parent station, so we don't know if a train is on the fence track, and it'd be a bit of work to change that. I guess the same might apply at Brattle  loop, too.

Reply:
If Glides is only looking at the parent station, how does it know that the train has turned around?

Reply:
By the direction we get from RTR. (Which, awkwardly, is a property of the trip assignment, hence the note about circular dependencies in the "Inputs" section.)

Reply:
@lemald or @paulswartz maybe you can chime in about how we think this will work on the RTR side? From an outsider perspective, it seems like the direction is part of the trip assignment, so Glides would need to be responsible for it (and couldn't rely on it as an input to the trip assignment).

Reply:
If I recall correctly, the direction assignment at RTR actually happens at a lower level of the logic than the trip assignment, the issue is just that trip details are the place where the direction ID actually flows through in the GTFS-rt, so it's more of an issue of how the data is formatted than a fundamental circular dependency in the logic.

Reply:
+1. Conceptually, the direction of a train can be defined by which track it's on and which way it last moved. That's important input information into the trip matching logic. If we want to remove the circular dependency in trip assignments, then we would need a separate non-trip direction field in GTFS-RT for RTR to communicate a train's direction to Glides.

We can completely prevent the logical circular dependency with these rules:
- Glides can use the direction from GTFS-RT's trip, but nothing else.
- RTR never uses Glides as input when determining a vehicle's direction. If it gets a trip assignment from Glides where the trip's direction doesn't match the direction it expects, then it ignores Glides' assignment and makes its own trip with the correct direction.